flink-clients/src/test/java/org/apache/flink/client/cli/CliFrontendStopWithSavepointTest.java

```diff
@@ -67,78 +62,119 @@ public void testStopWithOnlyJobId() throws Exception {
+			assertThat(jobID, is(jid));
+			assertThat(advanceToEndOfEventTime, is(false));
+			assertNull(savepointDirectory);
+			stopWithSavepointLatch.trigger();
+			return CompletableFuture.completedFuture(savepointDirectory);
+		});
 		MockedCliFrontend testFrontend = new MockedCliFrontend(clusterClient);
 
 		testFrontend.stop(parameters);
 
-		Mockito.verify(clusterClient, times(1))
-				.stopWithSavepoint(eq(jid), eq(false), isNull());
+		stopWithSavepointLatch.await();
 	}
 
 	@Test
 	public void testStopWithDefaultSavepointDir() throws Exception {
 		JobID jid = new JobID();
 
 		String[] parameters = {jid.toString() };
-		final ClusterClient<String> clusterClient = createClusterClient(null);
+		OneShotLatch stopWithSavepointLatch = new OneShotLatch();
+		TestingClusterClient<String> clusterClient = new TestingClusterClient<>();
+		clusterClient.setStopWithSavepointFunction((jobID, advanceToEndOfEventTime, savepointDirectory) -> {
+			assertThat(jobID, is(jid));
+			assertThat(advanceToEndOfEventTime, is(false));
+			assertNull(savepointDirectory);
+			stopWithSavepointLatch.trigger();
+			return CompletableFuture.completedFuture(savepointDirectory);
+		});
 		MockedCliFrontend testFrontend = new MockedCliFrontend(clusterClient);
 		testFrontend.stop(parameters);
 
-		Mockito.verify(clusterClient, times(1))
-				.stopWithSavepoint(eq(jid), eq(false), isNull());
+		stopWithSavepointLatch.await();
 	}
 
 	@Test
 	public void testStopWithExplicitSavepointDir() throws Exception {
 		JobID jid = new JobID();
 
 		String[] parameters = { "-p", "test-target-dir", jid.toString() };
-		final ClusterClient<String> clusterClient = createClusterClient(null);
+		OneShotLatch stopWithSavepointLatch = new OneShotLatch();
+		TestingClusterClient<String> clusterClient = new TestingClusterClient<>();
+		clusterClient.setStopWithSavepointFunction((jobID, advanceToEndOfEventTime, savepointDirectory) -> {
+			assertThat(jobID, is(jid));
+			assertThat(advanceToEndOfEventTime, is(false));
+			assertThat(savepointDirectory, is("test-target-dir"));
+			stopWithSavepointLatch.trigger();
+			return CompletableFuture.completedFuture(savepointDirectory);
+		});
 		MockedCliFrontend testFrontend = new MockedCliFrontend(clusterClient);
 		testFrontend.stop(parameters);
-
-		Mockito.verify(clusterClient, times(1))
-				.stopWithSavepoint(eq(jid), eq(false), eq("test-target-dir"));
+		stopWithSavepointLatch.await();
 	}
 
 	@Test
 	public void testStopOnlyWithMaxWM() throws Exception {
 		JobID jid = new JobID();
 
 		String[] parameters = { "-d", jid.toString() };
-		final ClusterClient<String> clusterClient = createClusterClient(null);
+		OneShotLatch stopWithSavepointLatch = new OneShotLatch();
+		TestingClusterClient<String> clusterClient = new TestingClusterClient<>();
+		clusterClient.setStopWithSavepointFunction((jobID, advanceToEndOfEventTime, savepointDirectory) -> {
+				assertThat(jobID, is(jid));
```

Comment:
Remove a tab from all these lines for uniformity.

Reply:
Thanks!